bestiary/core/db.py

```diff
@@ -328,6 +334,83 @@ def delete_project(trxl, project):
                        target=str(op_args['id']))
 
 
+def link_parent_project(trxl, project, parent_project):
+    """Link a project object with another with a child-parent relation.
+
+    This method sets the `parent_project` field value from `project` with the
+    `Project` object from `parent_project`. If this field already had a value,
+    it will be overwritten. In case the parent project is invalid, the method
+    will raise an `ValueError`.
+    The reasons for considering a parent as invalid are:
+      - The parent and the project are the same.
+      - The parent belongs to a different ecosystem.
+      - If the project is not a root one, when the parent comes from a
+       different root project.
+      - The parent is a descendant from the project.
+
+    :param trxl: TransactionsLog object from the method calling this one
+    :param project: Source project to be linked
+    :param parent_project: Parent project to be linked to the source project
+
+    :returns: the updated project object
+
+    :raises ValueError: raised either when the given parent project is invalid
+    """
+    def _get_root(project):
+        """Return the root project given one of its subprojects"""
+
+        parent = None
+        if project:
+            parent = project.parent_project
+        if parent:
+            project = _get_root(parent)
+
+        return project
+
+    def _is_descendant(project, from_project):
+        """Check if a project is a descendant of another"""
+
+        queue = [from_project]
+        while queue:
+            cur_project = queue.pop(0)
+            for subproject in cur_project.subprojects.all():
+                if subproject == project:
+                    return True
+                queue.append(subproject)
+
+    # Setting operation arguments before they are modified
+    op_args = {
+        'id': project.id,
+        'parent_id': parent_project.id
+    }
+
+    if project == parent_project:
+        raise ValueError('Project cannot be its own parent')
+    if project.ecosystem != parent_project.ecosystem:
+        raise ValueError('Parent cannot belong to a different ecosystem')
+    if _is_descendant(parent_project, project):
+        raise ValueError('Parent cannot be a descendant')
```

Comment:
I think we should also check that `parent_project != project`. Right now, the function is executed when there's no reason for.

Reply:
Done!

Reply:
Should you add this case to the documentation, I think it's not included.